tests/test_readers.py

```diff
@@ -179,6 +179,10 @@ def checkExposureFitsReader(self, exposureIn, fileName, dtypesOut):
         reader = ExposureFitsReader(fileName)
         self.assertIn('EXPINFO_V', reader.readMetadata().toDict(), "metadata is automatically versioned")
         reader.readMetadata().remove('EXPINFO_V')
+        self.assertIn('EXTNAME', reader.readMetadata().toDict(), "EXTNAME is added upon writing")
```

Comment:
I would really like a test that checks that EXTNAME has the expected value.

Reply:
I gave it a shot, but I'm pretty sure that's not what you had in mind. I essentially have a predetermined list of acceptable EXTNAMEs, and I doubt it is exhaustive (`PiffPsf` should probably be in there, though i'm not sure how we persist those and now that I think of it I doubt it's as part of Archives...?). If this is really kind of what we want, how can I make sure I've got every kind of Persistable we have?

It  works as is with all extensions from our regular processing calexps and coadds, fwiw.

I'm not sure how to be less stupid about this - since I can't know beforehand which HDU should correspond to which element.

Reply:
You do know the set of EXTNAMEs you expect for your specific test dataset don't you?

At the very least reading the file with astropy.io.fits and iterating over each HDU to check that it is returning an actual name for `.name` property seems useful. It would show that this ticket is doing something and would also break if somehow we stopped filling in a name.